source/common/upstream/original_dst_cluster.cc

```diff
@@ -0,0 +1,163 @@
+#include "common/upstream/original_dst_cluster.h"
+
+#include <chrono>
+#include <list>
+#include <string>
+#include <vector>
+
+#include "common/network/address_impl.h"
+#include "common/network/utility.h"
+
+namespace Envoy {
+namespace Upstream {
+
+OriginalDstCluster::LoadBalancer::LoadBalancer(HostSet& host_set, OriginalDstCluster& parent)
+    : host_set_(host_set), parent_(parent) {
+  // host_set_ is initially empty.
+  host_set_.addMemberUpdateCb([this](const std::vector<HostSharedPtr>& hosts_added,
+                                     const std::vector<HostSharedPtr>& hosts_removed) -> void {
+    // Update the hosts map
+    for (auto host : hosts_removed) {
+      ENVOY_LOG(debug, "REMOVING HOST {}", host->address()->asString());
```

Comment:
nit: please follow other logging which generally uses lower case (applies to all)